khelo/lib/ui/flow/score_board/components/select_player_sheet.dart

```diff
@@ -172,51 +177,53 @@ class _SelectPlayerSheetState extends ConsumerState<SelectPlayerSheet> {
-                type == PlayerSelectionType.batsMan &&
-                !isEnabled,
-            onTap: () => setState(() {
-              if (type == PlayerSelectionType.batsMan) {
-                if (widget.playerSelectionType == PlayerSelectionType.all) {
-                  if (batsMan1?.player.id == player.player.id) {
-                    batsMan1 = null;
-                  } else if (batsMan2?.player.id == player.player.id) {
-                    batsMan2 = null;
-                  } else if (batsMan1 == null) {
-                    batsMan1 = player;
-                  } else if (batsMan2 == null) {
-                    batsMan2 = player;
-                  } else {
-                    return;
-                    // extra else to suppress lint suggestion of using ??= for batsMan2 assignment
-                  }
-                } else {
+      children: list.map((player) {
+        final status = player.performance
+            .where((element) => element.inning_id == state.currentInning?.id)
+            .firstOrNull
+            ?.status;
+        return UserCellView(
+          title: player.player.name ?? context.l10n.common_anonymous_title,
+          imageUrl: player.player.profile_img_url,
+          initial: player.player.nameInitial,
+          subtitle: showOverCount
+              ? "(${context.l10n.match_list_overs_title(_getOverCount(state, player.player.id))})"
+              : null,
+          isSelected: type == PlayerSelectionType.batsMan
+              ? [batsMan1?.player.id, batsMan2?.player.id]
+                  .contains(player.player.id)
+              : bowler?.player.id == player.player.id,
+          tag: status == PlayerStatus.injured &&
+                  type == PlayerSelectionType.batsMan
+              ? context.l10n.score_board_injured_tag_title
+              : null,
+          disableCell: status == PlayerStatus.injured &&
+              type == PlayerSelectionType.batsMan &&
+              !isEnabled,
+          onTap: () => setState(() {
+            if (type == PlayerSelectionType.batsMan) {
+              if (widget.playerSelectionType == PlayerSelectionType.all) {
+                if (batsMan1?.player.id == player.player.id) {
+                  batsMan1 = null;
+                } else if (batsMan2?.player.id == player.player.id) {
+                  batsMan2 = null;
+                } else if (batsMan1 == null) {
                   batsMan1 = player;
+                } else if (batsMan2 == null) {
+                  batsMan2 = player;
+                } else {
+                  return;
+                  // extra else to suppress lint suggestion of using ??= for batsMan2 assignment
                 }
               } else {
-                bowler = player;
+                batsMan1 = player;
               }
-            }),
-          )
-        ],
-      ],
+            } else {
+              bowler = player;
+            }
+          }),
+        );
+      }).toList(),
```

Comment:
i think we need to simplify this conditions using creating variables.

Example:
`final isSelected=type == PlayerSelectionType.batsMan
              ? [batsMan1?.player.id, batsMan2?.player.id]
                  .contains(player.player.id)
              : bowler?.player.id == player.player.id;`